samcli/lib/runtimes/base.py

```diff
@@ -0,0 +1,342 @@
+import logging
+import os
+from dataclasses import dataclass, field
+from enum import Enum, unique
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, cast
+
+LOG = logging.getLogger(__name__)
+
+_init_path = Path(os.path.dirname(__file__)).parent.parent
+_templates = _init_path / "lib" / "init" / "templates"
+_lambda_images_templates = _init_path / "lib" / "init" / "image_templates"
+
+
+@unique
+class Architecture(Enum):
+    X86_64 = "x86_64"
+    ARM64 = "arm64"
+
+
+@dataclass
+class FamilyDataMixin:
+    key: str
+    layer_subfolder: Optional[str]
+    dep_manager: List[Tuple[str, Path]] = field(default_factory=list)  # (package manager, path to local init template)
+    build: bool = True
+    eb_code_binding: Optional[str] = None  # possible values are "Java8", "Python36", "Go1", "TypeScript3"
```

Comment:
What is this field for?

Reply:
This is for generating the EventBridge event code-binding

Reply:
I'm just not familiar with this? Why can it only be those values?

Reply:
https://docs.aws.amazon.com/eventbridge/latest/schema-reference/v1-registries-name-registryname-schemas-name-schemaname-language-language.html#PutCodeBinding

These are the enums / languages that EB Schema Code Binding supports.